### PR TITLE
Fixed unexpectable behavior of getAttributeRawValue(), issue #658

### PR DESCRIPTION
**Pull request by midlan**

### CODE REVIEW

File: app/code/core/Mage/Catalog/Model/Resource/Abstract.php [hunk 1]
```diff
@@ -584,8 +584,12 @@ public function getAttributeRawValue($entityId, $attribute, $store)
     {
         if (!$entityId || empty($attribute)) {
             return false;
-        }
+	}
```
Comment (colinmollenhour) [inline app/code/core/Mage/Catalog/Model/Resource/Abstract.php]: Looks like you replaced spaces with a tab.

File: app/code/core/Mage/Catalog/Model/Resource/Abstract.php [hunk 2]
```diff
@@ -584,8 +584,12 @@ public function getAttributeRawValue($entityId, $attribute, $store)
...
+
+        $wasArray = true;
```
Comment (colinmollenhour) [inline app/code/core/Mage/Catalog/Model/Resource/Abstract.php]: To better maintain BC I think `$wasArray` should also be false if the size of the `$attribute` array was 1. In this case perhaps it should be renamed to something like `$returnArray`.

File: app/code/core/Mage/Catalog/Model/Resource/Abstract.php (hunk 1 above)
Comment (midlan) [reply app/code/core/Mage/Catalog/Model/Resource/Abstract.php]: my fault, removed

File: app/code/core/Mage/Catalog/Model/Resource/Abstract.php (hunk 2 above)
Comment (midlan) [reply app/code/core/Mage/Catalog/Model/Resource/Abstract.php]: You are right, I didn't saw any code like that in Magento, but some 3rd party module maybe relying on it. Fixed that also.

**Pull request merged**